moshi/src/main/java/com/squareup/moshi/AdapterMethodsFactory.java

```diff
@@ -87,6 +87,10 @@ public AdapterMethodsFactory(List<AdapterMethod> toAdapters, List<AdapterMethod>
           }
         }
       }
+
+      @Override public String toString() {
+        return "JsonAdapter" + annotations + "(" + type + ")";
```

Comment:
It's a bit giant: `JsonAdapter[@com.squareup.moshi.JsonQualifiersTest$FooPrefix(), @com.squareup.moshi.JsonQualifiersTest$BazSuffix()](class java.lang.String)`